File: test_cli_output/horizon/scripts/EnvironmentController.ts
/**
           * Environment controller for world settings and lighting.
           * Weather: clear
           * Time: 12:00
           */

          import { World, Vec3, Color } from 'horizon/core';
          import { LightData, LightType } from './types';

          /**
           * Environment configuration.
           */
          export const ENVIRONMENT_CONFIG = {
            weather: "clear",
            timeOfDay: {
              hour: 12,
              minute: 0,
              dayNightCycle: false,
            },
            ambientColor: { r: 0.2, g: 0.2, b: 0.2, a: 1.0 },
            fogEnabled: false,
            fogColor: { r: 0.5, g: 0.5, b: 0.5, a: 1 },
            fogDensity: 0.01,
            skyboxType: "procedural",
          };

          /**
           * Light data for the world.
           */
          export const LIGHT_DATA: LightData[] = [
            {
  name: "Sun",
  lightType: LightType.Directional,
  color: { r: 1.0, g: 1.0, b: 1.0, a: 1 },
  intensity: 1.2,
  position: { x: 0.0, y: 10.0, z: 0.0 },
  rotation: { x: 50.0, y: -30.0, z: 0.0 },
  castShadows: true,
},
{
  name: "AmbientLight",
  lightType: LightType.Point,
  color: { r: 1.0, g: 1.0, b: 1.0, a: 1 },
  intensity: 0.5,
  position: { x: 0.0, y: 5.0, z: 0.0 },
  rotation: { x: 0.0, y: 0.0, z: 0.0 },
  castShadows: true,
}
          ];

          /**
           * Controller for environment and lighting settings.
           */
          export class EnvironmentController {
            private world: World;

            constructor() {
              this.world = World.getWorld();
            }

            /**
             * Setup complete environment.
             */
            async setup(): Promise<void> {
              console.log('Setting up environment...');

              this.setupAmbientLighting();
              this.setupFog();
              await this.createLights();

              console.log('Environment setup complete');
            }

            /**
             * Configure ambient lighting.
             */
            private setupAmbientLighting(): void {
              const { ambientColor } = ENVIRONMENT_CONFIG;
              // Horizon-specific ambient light setup
              console.log(`Ambient light: R${ambientColor.r} G${ambientColor.g} B${ambientColor.b}`);
            }

            /**
             * Configure fog settings.
             */
            private setupFog(): void {
              if (!ENVIRONMENT_CONFIG.fogEnabled) {
                return;
              }

              const { fogColor, fogDensity } = ENVIRONMENT_CONFIG;
              // Horizon-specific fog setup
              console.log(`Fog enabled: density=${fogDensity}`);
            }

            /**
             * Create all lights in the world.
             */
            private async createLights(): Promise<void> {
              for (const lightData of LIGHT_DATA) {
                await this.createLight(lightData);
              }
              console.log(`Created ${LIGHT_DATA.length} lights`);
            }

            /**
             * Create a single light.
             */
            private async createLight(data: LightData): Promise<void> {
              try {
                // Horizon-specific light creation
                console.log(`Creating light: ${data.name} (${data.lightType})`);
              } catch (error) {
                console.error(`Failed to create light ${data.name}:`, error);
              }
            }
          }

File: cli_test_output/unity/Scripts/EnvironmentController.cs
using UnityEngine;
using System.Collections.Generic;

namespace OmniWorld.Generated
{
    [System.Serializable]
    public class LightData
    {
        public string name;
        public LightType lightType;
        public Color color;
        public float intensity;
        public Vector3 position;
        public Vector3 rotation;
        public bool castShadows;
    }

    /// <summary>
    /// Controls environment settings for the generated world.
    /// </summary>
    public class EnvironmentController : MonoBehaviour
    {
        [Header("Environment Settings")]
        public Color ambientColor = new Color(0.2f, 0.2f, 0.2f, 1.0f);
        public bool fogEnabled = false;
        public Color fogColor = new Color(0.5f, 0.5f, 0.5f);
        public float fogDensity = 0.01f;

        [Header("Time Settings")]
        public int timeHour = 12;
        public int timeMinute = 0;
        public bool dayNightCycle = false;

        private List<Light> createdLights = new List<Light>();

        private LightData[] lightDataList = new LightData[]
        {
            new LightData {
        name = "Sun",
        lightType = LightType.Directional,
        color = new Color(1.0f, 1.0f, 1.0f),
        intensity = 1.2f,
        position = new Vector3(0.0f, 10.0f, 0.0f),
        rotation = new Vector3(50.0f, -30.0f, 0.0f),
        castShadows = true
    },
new LightData {
        name = "AmbientLight",
        lightType = LightType.Point,
        color = new Color(1.0f, 1.0f, 1.0f),
        intensity = 0.5f,
        position = new Vector3(0.0f, 5.0f, 0.0f),
        rotation = new Vector3(0.0f, 0.0f, 0.0f),
        castShadows = true
    }
        };

        public void Initialize()
        {
            SetupAmbient();
            SetupFog();
            CreateLights();
        }

        private void SetupAmbient()
        {
            RenderSettings.ambientLight = ambientColor;
            RenderSettings.ambientMode = UnityEngine.Rendering.AmbientMode.Flat;
        }

        private void SetupFog()
        {
            RenderSettings.fog = fogEnabled;
            RenderSettings.fogColor = fogColor;
            RenderSettings.fogDensity = fogDensity;
            RenderSettings.fogMode = FogMode.Exponential;
        }

        private void CreateLights()
        {
            foreach (var data in lightDataList)
            {
                CreateLight(data);
            }
            Debug.Log($"Created {lightDataList.Length} lights");
        }

        private void CreateLight(LightData data)
        {
            GameObject lightObj = new GameObject(data.name);
            lightObj.transform.position = data.position;
            lightObj.transform.rotation = Quaternion.Euler(data.rotation);
            lightObj.transform.SetParent(transform);

            Light light = lightObj.AddComponent<Light>();
            light.type = data.lightType;
            light.color = data.color;
            light.intensity = data.intensity;
            light.shadows = data.castShadows ? LightShadows.Soft : LightShadows.None;

            createdLights.Add(light);
        }
    }
}